### DISCUSSION TITLE
"-sketch" operator changed between v6 and v7

### DISCUSSION THREAD

**Discussion by AlexLaforge** [ImageMagick/ImageMagick#4306] [Digital Image Processing]
Hi,

I was used to apply the -sketch operator on images in ImageMagick v6.
The same command-line produces **very different (and not convincing) results** in v7 :

`convert wizard: -colorspace gray -sketch 0x40+135 C:\sketched.png`

How to obtain the same result in v7?

### V6
![v6](https://user-images.githubusercontent.com/8602659/135736193-e5aec0b2-5174-45a4-bd46-b2b13c07fabc.png)

### V7
![v7](https://user-images.githubusercontent.com/8602659/135736192-a81665ad-3c5c-464e-9154-a87d7bc56d6f.png)

**Comment by fmw42** [comment]
What exact versions of 6 and 7?

I get the following on IM 6.9.12-23 Mac OSX Sierra

`convert wizard: -colorspace gray -sketch 0x40+135 v6.png`

![v6](https://user-images.githubusercontent.com/13458218/135736587-96507c23-6341-4420-8fd9-e26f2746c315.png)

and on IM 7.1.0-8, I get

`magick wizard: -colorspace gray -sketch 0x40+135 v7.png`

![v7](https://user-images.githubusercontent.com/13458218/135736608-0c605e84-578b-4ada-b370-e02eda506e9b.png)

If you want it darker, then use linear gray:

`magick wizard: -colorspace lineargray -sketch 0x40+135 v7c.png
`
![v7c](https://user-images.githubusercontent.com/13458218/135736616-39d4c4c7-516a-493e-a916-049a0aa66eba.png)

**Reply by AlexLaforge** [reply]
The versions are **ImageMagick-6.8.5-Q16** vs **ImageMagick-7.1.0-Q8** on Windows Server 2012.

However, I just installed **ImageMagick-7.1.0-Q16** and it gave better results than its Q8 brother, close to the results obtained with the **ImageMagick-6.8.5-Q16**, but yes, I was forced to use `lineargray` instead of simply `gray`  ...

Why does 16 bits-per-pixel would provide so different results than the Q8?

**Reply by fmw42** [reply]
Less precision in computations. Q8 is limited to 256 values. Q16 would have 65535 values
